app/api/web/fetch_file.coffee

```diff
@@ -0,0 +1,87 @@
+#include all the required dependencies
+util = require('../../common/util')
+appconfig = require('../../config/appconfig').appconfig
+fs = require('fs')
+request = require('request')
+csrf = require('csurf')
+async = require('async')
+prettysize = require('prettysize')
+path = require('path')
+tmpDir = appconfig.getTemporaryDirectory()
+fileLimits = appconfig.getFileLimits()
+imageRegex = appconfig.getImageRegex()
+mimeRegex = appconfig.getMimeRegex()
+
+exports.init = (app) ->
+	
+	app.post '/fetch/', csrf({ cookie:true }), (req,res) ->
+
+		#set file name from req.body
+		fileName = path.basename(req.body.url)
+		#Remove unnecessary characters and spaces in the file
+		fileName = fileName.replace(/@[^0-9a-z\.]+@i/g, "-")
+		fileName = fileName.replace(/\ +/g, "-")
+		#give file a unique id ending with the file name , provided it has one.
+		fileName = "#{util.uniqueId()}-#{fileName}"
+		reqURL = [req.body.url]
```

Comment:
@gauravtiwari5050 I need your review here. Here the url of the target image is converted to an array in order to use `async.each`. Because we need the *response to be sent **only** when the **entire fetching and writing process is complete***. Please give suggestion for code improvement, if any.